Tests/CombineLatestManyTests.swift

```diff
@@ -0,0 +1,197 @@
+//
+//  CombineLatestManyTests.swift
+//  CombineExtTests
+//
+//  Created by Jasdev Singh on 3/22/20.
+//
+
+import Combine
+import CombineExt
+import XCTest
+
+final class CombineLatestManyTests: XCTestCase {
+    private var subscription: AnyCancellable!
+
+    private enum CombineLatestManyTestError: Error {
+        case anError
+    }
+
+    func testCollectionCombineLatestWithFinishedEvent() {
+        let first = PassthroughSubject<Int, Never>()
+        let second = PassthroughSubject<Int, Never>()
+        let third = PassthroughSubject<Int, Never>()
+        let fourth = PassthroughSubject<Int, Never>()
+
+        var completed = false
+        var results = [[Int]]()
+
+        subscription = [first, second, third, fourth]
+            .combineLatest()
+            .sink(receiveCompletion: { _ in completed = true },
+                  receiveValue: { results.append($0) })
+
+        first.send(1)
+        second.send(2)
+
+        XCTAssertTrue(results.isEmpty)
+        XCTAssertFalse(completed)
+
+        third.send(3)
+        fourth.send(4)
+
+        XCTAssertEqual(results, [[1, 2, 3, 4]])
+        XCTAssertFalse(completed)
+
+        first.send(1)
+
+        XCTAssertEqual(results, [[1, 2, 3, 4], [1, 2, 3, 4]])
+        XCTAssertFalse(completed)
+
+        fourth.send(4)
```

Comment:
it would be better if the updated values you're sending are different in their respective indices, so it's obvious a change was made IMO